test/drenv/drenv_test.py

```diff
@@ -79,10 +78,7 @@ def test_missing_addon(tmpdir):
     path = tmpdir.join("missing-addon.yaml")
     path.write(content)
 
-    warning = r"WARNING .+ 'no-such-addon'"
-
     # Use subprocess.run() to get access to stderr on success.
```

Comment:
This comment is stale now - but good that you kept it since it shows that we don't need to use subprocess.run. We can simplify this test using commands.run().

```
--- a/test/drenv/drenv_test.py
+++ b/test/drenv/drenv_test.py
@@ -76,9 +76,7 @@ profiles:
           - name: no-such-addon
 """
     path = tmpdir.join("missing-addon.yaml")
     path.write(content)
 
-    # Use subprocess.run() to get access to stderr on success.
-    cp = subprocess.run(["drenv", "start", str(path)], stderr=subprocess.PIPE)
-
-    assert cp.returncode == 1
+    with pytest.raises(commands.Error):
+        commands.run("drenv", "start", str(path))
```